components/kyma-environment-broker/common/gardener/client.go

```diff
@@ -4,56 +4,68 @@ import (
 	"fmt"
 	"io/ioutil"
 
-	gardener_apis "github.com/gardener/gardener/pkg/client/core/clientset/versioned/typed/core/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	restclient "k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-func NewGardenerClusterConfig(kubeconfigPath string) (*restclient.Config, error) {
+type SecretBinding struct {
+	unstructured.Unstructured
+}
 
-	rawKubeconfig, err := ioutil.ReadFile(kubeconfigPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read Gardener Kubeconfig from path %s: %s", kubeconfigPath, err.Error())
-	}
+func (b SecretBinding) GetSecretRefName() string {
+	//TODO: implement jw check errors
```

Comment:
wondering if it's worth to return the error and adjust the callers of this to deal with it or if we should straight panic here?

Reply:
after a discussion with @szwedm, putting `panic()` here. It's near impossible this will ever happen but it's better that we don't allow the code to continue in that case.